### ISSUE TITLE
Error when leaving Gutenberg page

### ISSUE THREAD

**Issue by dimalif** [WordPress/gutenberg#27696]
Wordpress 5.5.3 & 5.6 - PHP 7.3 or 7.4

When editing a page (or just viewing without changes), i get the following error in my browser console. I have tried and searched everything i could, no luck.

Any ideas what this could be?

```
Uncaught (in promise) 
Object { code: "fetch_error", message: "You are probably offline." }
data.min.js:2:29815
    n https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    a https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    (Async: promise callback)
	
    n https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    c https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    o https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    o https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    s https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    a https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    u https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
	
    pi https://www.website.com/wp-includes/js/dist/editor.min.js?ver=a65f9b8727c91b9df1641cfd0545e051:12
    n https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    Fe https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    Ve https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    je https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:78
    ph https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:215
    lh https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:126
    O https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:121
    O self-hosted:1265
	
    ze https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:118
    ze self-hosted:1265
	
    mg https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:53
    unstable_runWithPriority https://www.website.com/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26
    Ma https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
	
    mg https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    V https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    Sb https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:213
    Mg https://www.website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:85
    Mg self-hosted:1265
	
    e https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    t https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    o https://www.website.com/wp-includes/js/dist/data.min.js?ver=75f90354ddff4acd5b0b4026454037ca:2
    forEach self-hosted:
```

**Comment by dimalif** [comment]
Keep in mind that i can save properly, no errors and also i haven't any "mixted" content like i saw reported from an other user. Just this error when i leave a Gutenberg page.

**Comment by talldan** [comment]
@dimalif Would you be able to share which browser you're using?

It seems like an HTTP request is interrupted by you leaving the page, but it's unusual to always have a request in progress.

It could be autosave, do you know if your site makes any changes to the AUTOSAVE_INTERVAL?

If you're comfortable with the browser developer tools, then checking there to see if there are any pending network requests could be useful.

**Comment by dimalif** [comment]
I'm using Firefox and Chrome. 

Autosave was default when the issue happens, i changed it to 3, still no luck.

There are no pending requests from what i noticed during the checks that this seems to happen when i exit **quickly** Guterberg.

When i stay in a page **for a few seconds more** and then exit, it shows no errors.

**Comment by dimalif** [comment]
@talldan 

Doing some more checks as you suggested for network pending requests, i saw sometimes this different json error:
```

Uncaught (in promise) 
Object { code: "invalid_json", message: "The response is not a valid JSON response." }
data.min.js:2:7527
    n https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    a https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    (Async: promise callback)
    n https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    c https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    o https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    o https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    (Async: setTimeout handler)
    e https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    tryCatch https://website.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.4.4:1
    invoke https://website.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.4.4:1
    n https://website.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.4.4:1
    n https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    c https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    o https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    o https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    f https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    s https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    c https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    r https://website.com/wp-includes/js/dist/edit-post.min.js?ver=f03e57452d64a3c73916cb7f8470bf5c:2
    We https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    wr https://website.com/wp-includes/js/dist/edit-post.min.js?ver=f03e57452d64a3c73916cb7f8470bf5c:2
    we https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:84
    zj https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:226
    Th https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:152
    tj https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:152
    Te https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:146
    Te self-hosted:1208
    Pg https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:61
    unstable_runWithPriority https://website.com/wp-includes/js/dist/vendor/react.min.js?ver=16.13.1:25
    Da https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:60
    Pg https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:61
    ha https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:60
    Ja https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:224
    ch https://website.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:93
    ch self-hosted:1265
    e https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
    t https://website.com/wp-includes/js/dist/data.min.js?ver=777f925bfe1a2aea2a86b7b9ae6aed25:2
```

**Comment by dimalif** [comment]
I updated to 5.7 and i still experience this slightly different issue. Trying to trash a page, i get:

Uncaught (in promise) 
Object { code: "fetch_error", message: "You are probably offline." },  redux-routine.min.js:2:5160

**Comment by dimalif** [comment]
Also the issues are using Firefox 64 latest and no addons. 

I was wrong that with Chrome there are no issues, the log was just scrolled.

api-fetch.min.js?ver=a783d1f442d2abefc7d6dbd156a44561:2 GET https://www . website . com/wp-admin/post.php?post=10964&action=edit&message=4 500

and

redux-routine.min.js?ver=284288e957394d2097c6fbe95625e2fb:2 Uncaught (in promise) Response {type: "basic", url: "https://www. website . com /wp-admin/post.php?post=10964&action=edit&message=4", redirected: true, status: 500, ok: false, …}

**Comment by dimalif** [comment]
Is that "500" i see in status an error 500 (internal error 500) ? If it is, i could ask my hosting company for more details? 

Can someone verify this "500" is an internal error?

**Comment by skorasaurus** [comment]
A 500 error can be caused [by many different things ](https://kinsta.com/blog/500-internal-server-error/).

The link above has several strategies that you can try to resolve this. 

Also, Are you able to reproduce this after you have installed and activated the newest version of Gutenberg that is available through [the Gutenberg plugin](https://wordpress.org/plugins/gutenberg/)?

**Comment by dimalif** [comment]
@skorasaurus 

This happens within Gutenberg and nowhere else. And only under those circustances. I use WP latest but not with the plugin. 

Anyways this error doesn't seem to do any harm, i'm able to do what i need to do. 

My hosting provider pointed to WP and Gutenberg and they see no error in their logs. And since they are a small and dedicated company, they always have been right the last 8 years i use them. They fix their issues and point to something else if it is not their problem.

**Comment by annezazu** [comment]
@dimalif Thanks for the additional context. Based on these error messages and your description of your hosting environment, this seems to be an error in using WordPress on a likely smaller host with less dedicated resources per site. As @skorasaurus pointed out, a 500 error can be due to many different things! If you're able to work with your host and they indicate a deeper problem with Gutenberg, please re-open this so we can investigate further. At this time though, without any additional reports, it's likely that this is an individual issue with your site rather than a project wide problem.

**Comment by dimalif** [comment]
This happens to each and every WP installation i have tried in each and every hosting i tried (of customers accounts). And i was not aware that WP needed a super computer to run. Oh 'com on. guys. Fix your issues and stop bumping the versions.

Sorry to see the hosting blaim card played. Instead of trying to replicate it. 

pathetic.

**Comment by annezazu** [comment]
@dimadin I did try to replicate this across multiple different test sites and couldn't do so. I also test Gutenberg nearly every day and have never ran into this across any hosting environment I have. 

While responding to you and trying to rack my brain for any additional possible causes, I remembered this issue which has a great explanation of what's likely going on: https://github.com/WordPress/gutenberg/issues/23942#issuecomment-692491244 As noted there:

>If the subsites address is edited and changed to https instead everything starts to work as expected since mixed-content issues are avoided/resolved.

I truly hope this helps. I recommend following the related core issue as well: https://core.trac.wordpress.org/ticket/27499
